File: Python/rules/seatbelt_rule.py
from rules.rule import Rule
from model.violation import Violation

class SeatbeltRule(Rule):
    def __init__(self, fee):
        self.fee = fee

    def evaluate(self, observation):
        if not observation.is_seatbelt_fastened():
            return Violation("Seatbelt not fastned", self.fee)
        return None

File: JAVA/rules/SeatbeltRule.java
package rules;
import model.CarObservation;
import model.Violation;
import java.util.Optional;

public class SeatbeltRule implements Rule {
    private final double fee;
    public SeatbeltRule(double fee) {
        this.fee = fee;
    }
    @Override
    public Optional<Violation> evaluate(CarObservation observation) {
        if (!observation.isSeatbeltFastened()) {
            return Optional.of(new Violation("Seatbelt not fastened", fee));
        }
        return Optional.empty();
    }
}